src/string_radix_sort.jl

```diff
@@ -0,0 +1,233 @@
+            end
+            lens -= 16
+        elseif lens > 4
+            skipbytes = max(0, skipbytes - 8)
+            if o == Reverse
+                sorttwo!(.~load_bits.(UInt64, svec, skipbytes), svec)
+            else
+                sorttwo!(load_bits.(UInt64, svec, skipbytes), svec)
+            end
+            lens -= 8
+        else
+            skipbytes = max(0, skipbytes - 4)
+            if o == Reverse
+                sorttwo!(.~load_bits.(UInt32, svec, skipbytes), svec)
+            else
+                sorttwo!(load_bits.(UInt32, svec, skipbytes), svec)
+            end
+            lens -= 4
+        end
+    end
+    svec
+end
+
+"""
+    sorttwo!(vs, index)
+
+Sort both the `vs` and reorder `index` at the same. This allows for faster sortperm
+for radix sort.
+"""
+function sorttwo!(vs::AbstractVector{T}, index, lo::Int = 1, hi::Int=length(vs)) where T
+    # Input checking
+    if lo >= hi;  return (vs, index);  end
+
+    # Make sure we're sorting a bits type
+    # T = Base.Order.ordtype(o, vs)
+    if !isbits(T)
+        error("Radix sort only sorts bits types (got $T)")
+    end
+    o = Forward
+
+    # Init
+    iters = ceil(Integer, sizeof(T)*8/RADIX_SIZE)
+    bin = zeros(UInt32, 2^RADIX_SIZE, iters)
+    if lo > 1;  bin[1,:] = lo-1;  end
+
+    # Histogram for each element, radix
+    for i = lo:hi
+        v = uint_mapping(o, vs[i])
+        for j = 1:iters
+            idx = Int((v >> (j-1)*RADIX_SIZE) & RADIX_MASK) + 1
+            @inbounds bin[idx,j] += 1
+        end
+    end
+
+    # Sort!
+    swaps = 0
+    len = hi-lo+1
+
+    index1 = similar(index)
+    ts=similar(vs)
+    for j = 1:iters
+        # Unroll first data iteration, check for degenerate case
+        v = uint_mapping(o, vs[hi])
+        idx = Int((v >> (j-1)*RADIX_SIZE) & RADIX_MASK) + 1
+
+        # are all values the same at this radix?
+        if bin[idx,j] == len;  continue;  end
+
+        cbin = cumsum(bin[:,j])
+        ci = cbin[idx]
+        ts[ci] = vs[hi]
+        index1[ci] = index[hi]
+        cbin[idx] -= 1
+
+        # Finish the loop...
+        @inbounds for i in hi-1:-1:lo
+            v = uint_mapping(o, vs[i])
+            idx = Int((v >> (j-1)*RADIX_SIZE) & RADIX_MASK) + 1
+            ci = cbin[idx]
+            ts[ci] = vs[i]
+            index1[ci] = index[i]
+            cbin[idx] -= 1
+        end
+        vs,ts = ts,vs
+        index, index1 = index1, index
+        swaps += 1
+    end
+
+    if isodd(swaps)
+        vs,ts = ts,vs
+        index, index1 = index1, index
+        for i = lo:hi
+            @inbounds vs[i] = ts[i]
+            @inbounds index[i] = index1[i]
+        end
+    end
+    (vs, index)
+end
+
+"""
+    sortperm_radixsort()
+
+Currently radixsot 
+"""
+function sortperm_radixsort(svec::AbstractVector{String}; rev::Union{Bool,Void}=nothing, order::Ordering=Forward)
+    siv = StringIndexVector(copy(svec), collect(1:length(svec)))
```

Comment:
Usually you would define a `sortperm_radixsort!` function that doesn't make a copy of `svec`, and then define `sortperm_radixsort(svec, ...) = sortperm_radixsort!(copy(svec), ...)`

Reply:
done